rpc/typed_api.go

```diff
@@ -17,11 +17,66 @@ package rpc
 import (
 	"context"
 	"fmt"
+	"time"
 
 	"github.com/livekit/protocol/livekit"
+	"github.com/livekit/protocol/logger"
+	protopsrpc "github.com/livekit/protocol/psrpc"
+	"github.com/livekit/protocol/utils"
 	"github.com/livekit/psrpc"
+	"github.com/livekit/psrpc/pkg/middleware"
 )
 
+type PSRPCConfig struct {
+	Enabled     bool          `yaml:"enabled,omitempty"`
+	MaxAttempts int           `yaml:"max_attempts,omitempty"`
+	Timeout     time.Duration `yaml:"timeout,omitempty"`
+	Backoff     time.Duration `yaml:"backoff,omitempty"`
+	BufferSize  int           `yaml:"buffer_size,omitempty"`
+}
+
+var DefaultPSRPCConfig = PSRPCConfig{
+	MaxAttempts: 3,
+	Timeout:     500 * time.Millisecond,
+	Backoff:     500 * time.Millisecond,
+	BufferSize:  1000,
+}
+
+type ClientParams struct {
+	PSRPCConfig
+	Bus              psrpc.MessageBus
+	Logger           logger.Logger
+	Observer         middleware.MetricsObserver
+	VersionGenerator utils.TimedVersionGenerator
+}
+
+func NewClientParams(
+	config PSRPCConfig,
+	bus psrpc.MessageBus,
+	logger logger.Logger,
+	observer middleware.MetricsObserver,
+) ClientParams {
```

Comment:
No version generator? The structure above has that.

Reply:
no, i missed this, thanks 👍